core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -151,9 +151,21 @@ END_FUNC wait_secondary
 LOCAL_FUNC reset , :
 UNWIND(	.fnstart)
 UNWIND(	.cantunwind)
+
+/*
+ * if built-in arguments are configured, no arguments are expected from
+ * loader; instead, required parameters are configured in build time.
+ */
+
+#if defined (CFG_TEE_BUILT_IN_ARGS)
+    ldr r4, =CFG_TEE_PAGEABLE_ADDR
```

Comment:
I don't object to optional hardcoding of these addresses as it doesn't impact OP-TEE much, but I suspect it will cause occasional problems when they are changed. Since something has already has put stuff at those addresses, why not supply that information in registers?